app/components/works/buttons_component.rb

```diff
@@ -46,5 +46,12 @@ def cancel_link_location
         collection_works_path(collection)
       end
     end
+
+    def validate_user_version?
+      return false unless Settings.user_versions_ui_enabled
+      return true if %w[deposited rejected version_draft].include? work_version.state
```

Comment:
Can we just return this (and remove the `false` at the end?
```suggestion
  %w[deposited rejected version_draft].include? work_version.state
```